### PR TITLE
ignore case of `with header row` in sql when creating external table

### PR DESCRIPTION
**Pull request by lichuan6**
Resolves https://github.com/apache/arrow-datafusion/issues/1242

Currently, creating an external table using `datafusion-cli` with this sql:

```SQL
CREATE EXTERNAL TABLE t(c1 int) STORED AS CSV with header row LOCATION 'foo.csv'
```
will fail when `with header row` tokens are in lower case. This pr is a fix.

### CODE REVIEW

File: datafusion/src/sql/parser.rs [hunk 1]
```diff
@@ -294,7 +294,7 @@ impl<'a> DFParser<'a> {
     }
 
     fn consume_token(&mut self, expected: &str) -> bool {
```
[the remark this answers is not in the document]
Comment (lichuan6) [reply datafusion/src/sql/parser.rs]: fixed :)

**Review by alamb — APPROVED** [review]
Looks good to me. Thank you @lichuan6

**Comment by alamb** [comment]
@lichuan6  there appears to be some failures on the CI checks

**Comment by lichuan6** [comment]
> @lichuan6 there appears to be some failures on the CI checks

OK, thanks for pointing it out. It seems using `self.consume_token(&Token::make_keyword("WITH"))` will only recognize upper case keyword. What's the best way to do case-insensitive token parsing?

**Comment by lichuan6** [comment]
@Dandandan Could you please approve and make the tests run?

**Review by alamb — APPROVED** [review]
Looks good to me. Thank you @lichuan6 

Any thoughts @Dandandan ?

**Comment by woodsmur** [comment]
@Dandandan Could you please review this PR?

**Comment by lichuan6** [comment]
Hi, @Dandandan Could you please review this PR?

**Comment by alamb** [comment]
If @Dandandan  doesn't have anything to add I'll plan to merge this PR tomorrow

**Comment by alamb** [comment]
Thanks again @lichuan6

**Pull request merged**